src/plugins/esProxyAuth/index.js

```diff
@@ -0,0 +1,46 @@
+let _ = require('lodash');
+let btoa = require('btoa');
+let Promise = require('bluebird');
+let strategyMap = require('../../server/auth/lib/authStrategyMap');
+let name = 'esProxy';
+
+module.exports = (kibana) => new kibana.Plugin({
+  require: ['elasticsearch'],
+  init(server, options) {
+    let config = server.config();
+    if (!config.get('auth.enabled') || config.get('auth.strategy') !== name) return;
```

Comment:
I think these plugins should just always provided their strategy to the `strategyMap` and let the strategy consumers worry about their own selection criteria.